crates/optimism/node/src/node.rs

```diff
@@ -178,12 +179,20 @@ where
         OpAddOns<NodeAdapter<N, <Self::ComponentsBuilder as NodeComponentsBuilder<N>>::Components>>;
 
     fn components_builder(&self) -> Self::ComponentsBuilder {
-        let Self { args } = self;
-        Self::components(args.clone())
+        Self::components(&self.args)
     }
 
     fn add_ons(&self) -> Self::AddOns {
-        OpAddOns::new(self.args.sequencer_http.clone())
+        let mut builder = OpAddOns::builder();
+        if let Some(sequencer) = &self.args.sequencer_http {
+            builder = builder.with_sequencer(sequencer.clone());
+        }
+
+        if !self.args.storage_proof_only.is_empty() {
+            builder = builder.with_storage_proof_only(self.args.storage_proof_only.clone());
+        }
+
+        builder.build()
```

Comment:
this should be a chain of `with_` calls instead, we can make the args on builder optional instead